diff --git a/build_kernel.sh b/build_kernel.sh
index 671cf12..0f1ef4c 100755
--- a/build_kernel.sh
+++ b/build_kernel.sh
@@ -199,7 +199,7 @@ echo "debug: CC=${CC}"
 . ${DIR}/version.sh
 export LINUX_GIT
 
-#unset FULL_REBUILD
+unset FULL_REBUILD
 FULL_REBUILD=1
 if [ "${FULL_REBUILD}" ] ; then
 	/bin/sh -e "${DIR}/scripts/git.sh" || { exit 1 ; }
@@ -211,6 +211,15 @@ if [ "${FULL_REBUILD}" ] ; then
 	patch_kernel
 	copy_defconfig
 fi
+
+cd KERNEL
+patch -p1 < ../../xenomai-2.6.3/ksrc/arch/arm/patches/beaglebone/ipipe-core-3.8.13-beaglebone-pre.patch
+
+../../xenomai-2.6.3/scripts/prepare-kernel.sh --arch=arm --adeos=../../xenomai-2.6.3/ksrc/arch/arm/patches/ipipe-core-3.8.13-arm-3.patch --linux=.
+
+patch -p1 < ../../xenomai-2.6.3/ksrc/arch/arm/patches/beaglebone/ipipe-core-3.8.13-beaglebone-post.patch
+cd ..
+
 if [ ! ${AUTO_BUILD} ] ; then
 	make_menuconfig
 fi
diff --git a/version.sh b/version.sh
index 108279a..5e29fdb 100644
--- a/version.sh
+++ b/version.sh
@@ -21,8 +21,8 @@ config="omap2plus_defconfig"
 
 #linaro_toolchain="arm9_gcc_4_7"
 #linaro_toolchain="cortex_gcc_4_6"
-linaro_toolchain="cortex_gcc_4_7"
-#linaro_toolchain="cortex_gcc_4_8"
+#linaro_toolchain="cortex_gcc_4_7"
+linaro_toolchain="cortex_gcc_4_8"
 
 #Kernel/Build
 KERNEL_REL=3.8
